python/tests/lis/test_load.py

```diff
@@ -0,0 +1,586 @@
+import os
+import pytest
+
+from dlisio import lis
+
+def test_partitioning_1reel_1tape(tmpdir, merge_lis_prs):
+    # A well formatted, simple file
+    #
+    # file.lis
+    # |
+    # |->Reel 1
+    #    |-> Tape 1
+    #        |-> Logical File 1
+    #        |-> Logical File 2
+    #
+    fpath = os.path.join(str(tmpdir), '1reel1tape.dlis')
+    content = [
+        'data/lis/records/RHLR-1.lis.part',
+        'data/lis/records/THLR-1.lis.part',
+        'data/lis/records/FHLR-1.lis.part',
+        'data/lis/records/FTLR-1.lis.part',
+        'data/lis/records/FHLR-2.lis.part',
+        'data/lis/records/FTLR-2.lis.part',
+        'data/lis/records/TTLR-1.lis.part',
+        'data/lis/records/RTLR-1.lis.part',
+    ]
+    merge_lis_prs(fpath, content)
+
+    with lis.load(fpath) as (f1, f2, *tail):
+        assert len(tail) == 0
+        #  Check that f1, f2 is partitioned correctly by investigating their
+        #  FHLR/FTLR.
+        #  Check that f1, f2 "belongs" to the correct Reel and Tape by
+        #  investigating the assigned RHLR/RTLR and THLR/TTLR. The fields
+        #  checked are arbitrary as long as they can uniquely identify the
+        #  record.
```

Comment:
1. f1, f2 (are) partitioned correctly (xN)
2. f1, f2 "belong()" (xN)
3. I am afraid you won't be able to parametrize it nicely as you wanted because you have different file structures and it's big. I don't actually see any other pretty solution.
From where I sit on the sofa now, the best option seems to only put the check itself into separate function:
```
expected = Expected() #as default
expected.f1.set.reel.header(1).trailer(None) #whats not default
do_the_checks(f1, f2, expected)
```
But I would be fine with whatever solution you end up with. We can switch if we figure out a nicer solution later.

Reply:
Gonna leave as-is. I had a thought for how to parametrise, but it was not as simple as I imaged, so I forfeited the idea as the effort overweights the benefit